### PR TITLE
Issue #218: Add a hostname to be used as a class member in all the extensions

### PR DESCRIPTION
**Pull request by CherfaElyes**
* Added a getHostname() method to the TelemetryManager, which allows to return the hostname from a specific IConfiguration classes list, otherwise from the TelemetryManager.
* Added a hostname for each IConfiguration.
* Refactored all the Extensions to use the new getHostname() method.
* Refactored unit tests to make them run again.
* Tested all the protocols using MetricsHub engine.
* Documented the changes.

### CODE REVIEW

File: metricshub-doc/src/site/markdown/configuration/configure-agent.md [hunk 1]
```diff
@@ -872,3 +882,31 @@ Timeouts, durations and periods are specified with the below format:
 | m    | minutes                         | 90m, 1m15s       |
 | h    | hours                           | 1h, 1h30m        |
 | d    | days (based on a 24-hour day)   | 1d               |
+
+## Customise the hostname
+
+By default, the `host.name` attribute is used as:
+1. The hostname or IP address of the resource
+2. A hostname on each metric in Prometheus.
```
Comment (NassimBtk) [inline metricshub-doc/src/site/markdown/configuration/configure-agent.md:890]: `2. The hostname of each OpenTelemetry metric attached to the host resource.`

File: metricshub-doc/src/site/markdown/configuration/configure-agent.md [hunk 2]
```diff
@@ -872,3 +882,31 @@ Timeouts, durations and periods are specified with the below format:
...
 | h    | hours                           | 1h, 1h30m        |
 | d    | days (based on a 24-hour day)   | 1d               |
+
+## Customise the hostname
+
+By default, the `host.name` attribute is used as:
+1. The hostname or IP address of the resource
+2. A hostname on each metric in Prometheus.
+
+If the `hostname` parameter is specified in the protocol configuration, it overrides the `host.name` attribute for client requests. The `host.name` will be used only as a metric hostname in Prometheus.
```
Comment (NassimBtk) [inline metricshub-doc/src/site/markdown/configuration/configure-agent.md:892]: If the `hostname` parameter is specified in the protocol configuration, it overrides the `host.name` attribute for client requests. The `host.name` will be used only as a metric attribute.

File: metricshub-snmp-extension-common/src/test/java/org/sentrysoftware/metricshub/extension/snmp/source/SnmpGetSourceProcessorTest.java [hunk 3]
```diff
@@ -88,12 +89,14 @@ public void testProcess_WhenRequestExecutorReturnsValidResult_ReturnsSourceTable
 		assertEquals("result", result.getTable().get(0).get(0));
 	}
 
-	//Utility method to create a simulated TelemetryManager
-	private TelemetryManager mockTelemetryManager() {
-		TelemetryManager telemetryManager = mock(TelemetryManager.class);
-		HostConfiguration hostConfigurationMock = mock(HostConfiguration.class);
-		when(telemetryManager.getHostConfiguration()).thenReturn(hostConfigurationMock);
-		when(hostConfigurationMock.getHostname()).thenReturn("hostname");
+	// Utility method to create a telemetryManager
```
Comment (NassimBtk) [inline metricshub-snmp-extension-common/src/test/java/org/sentrysoftware/metricshub/extension/snmp/source/SnmpGetSourceProcessorTest.java:92]: Can we fix the javadoc here please.

File: metricshub-snmp-extension-common/src/test/java/org/sentrysoftware/metricshub/extension/snmp/source/SnmpTableSourceProcessorTest.java [hunk 4]
```diff
@@ -126,12 +127,14 @@ public void testProcess_WhenExecutorReturnsValidResult_ReturnsSourceTableWithRes
...
 	}
 
-	// Utility method to simulate a TelemetryManager
-	private TelemetryManager mockTelemetryManager() {
-		TelemetryManager telemetryManager = mock(TelemetryManager.class);
-		HostConfiguration hostConfigurationMock = mock(HostConfiguration.class);
-		when(telemetryManager.getHostConfiguration()).thenReturn(hostConfigurationMock);
-		when(hostConfigurationMock.getHostname()).thenReturn("hostname");
+	// Utility method to create a telemetryManager
+	private TelemetryManager createTelemetryManagerWithHostConfiguration() {
```
Comment (NassimBtk) [inline metricshub-snmp-extension-common/src/test/java/org/sentrysoftware/metricshub/extension/snmp/source/SnmpTableSourceProcessorTest.java:131]: Java doc is better ;)

File: metricshub-win-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/win/detection/WmiCriterionProcessor.java [hunk 5]
```diff
@@ -192,7 +193,7 @@ public NamespaceResult detectNamespace(
 					// This error indicates that the CIM server will probably never respond to anything
 					// (timeout, or bad credentials), so there's no point in pursuing our efforts here.
 					log.debug(
-						"Hostname %s - Does not respond to %s requests. %s: %s\nCancelling namespace detection.",
+						"Hostname {} - Does not respond to {} requests. {}: {}\nCancelling namespace detection.",
```
Comment (NassimBtk) [inline metricshub-win-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/win/detection/WmiCriterionProcessor.java:196]: Thank you 🤔

**Review by NassimBtk — CHANGES_REQUESTED** [review]

File: metricshub-win-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/win/detection/WmiCriterionProcessor.java (hunk 5 above)
Comment (CherfaElyes) [reply metricshub-win-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/win/detection/WmiCriterionProcessor.java:196]: My pleasure

**Review by NassimBtk — APPROVED** [review]

**Pull request merged**